python/lsst/resources/_resourcePath.py

```diff
@@ -85,11 +85,14 @@ class ResourcePath:
     root : `str` or `ResourcePath`, optional
         When fixing up a relative path in a ``file`` scheme or if scheme-less,
         use this as the root. Must be absolute.  If `None` the current
-        working directory will be used. Can be a file URI.
+        working directory will be used. Can be a file URI. Not used if
+        ``forceAbsolute`` is `False`.
```

Comment:
What happens if `uri` is a relative with `file` scheme (`file:a/b.c`) but `root` uses a different scheme? Does it make sense to support something like this, or will it fail with an exception?

Reply:
`file:a/b.c` is not really even allowed -- it's not a valid URI but `urllib.parse` does support it. I currently convert it to an absolute URI because a `file` URI must be an absolute path and now that scheme-less implies "relative path" and can be relative to anything I either have to convert this to an absolute file URI or raise an exception because I don't have a "relative schemeless that MUST BE A FILE" any more.